fract4d/c/model/site.cpp

```diff
@@ -96,6 +93,7 @@ void FDSite::status_changed(int status_val)
 bool FDSite::is_interrupted()
 {
     //fprintf(stderr,"int: %d\n",interrupted);
+    const std::lock_guard<std::mutex> lock(interrupt_lock);
```

Comment:
don't quite understand why you need a mutex here. We're just checking the current value of a bool. Nothing bad happens if we don't detect that we were interrupted until the next time we check.

Reply:
I think it’s because of the volatile qualifier that threadsanitizer complains.
Didn’t want to remove the qualifier, I’m not sure I’m missing something about the reason to have it. But maybe it makes sense to just remove it.

Reply:
interrupted should most probably be std::atomic<> and that makes the mutex and data race go away. Volatile is not the right language feature in this case as it doesn't mean thread safe and only causes the compiler to always emit read instructions when the variable is referenced, it doesn't make the compiler ensure cache cohearency or atomicity.